### ISSUE TITLE
Unix build - Boost isn't optional

### ISSUE THREAD

**Issue by Tezirg** [Qihoo360/evpp#204]
Hello, I am using Ubuntu 18

Great work guys, evpp is really well designed. I've noticed that evpp is advertised as a c++11 library but it is impossible to build it without installing boost. 

They are different build targets to compile both versions with and without boost but there are not options to disable the boost build. Therefore, it is not optional as advertised.. https://github.com/Qihoo360/evpp/blob/master/evpp/CMakeLists.txt#L33

In the same fashion, there are no options to disable the building of tests, apps, examples and benchmarks. 
https://github.com/Qihoo360/evpp/blob/master/CMakeLists.txt#L93

Would it be possible to add such options in the future ? Like libevent does ? 

For example, we could imagine a set of CMake options like: 
 `EVPP__DISABLE_BOOST ` `EVPP__DISABLE_TESTS` `EVPP__DISABLE_BENCHMARK`

libevent options example:
https://github.com/libevent/libevent/blob/master/CMakeLists.txt#L127
libevent conditional build example: 
https://github.com/libevent/libevent/blob/master/CMakeLists.txt#L966

Cheers

**Comment by gocpplua** [comment]
Yes, when I want to compile a version that does not apply to boost, it keeps failing
